### PR TITLE
[marl] update to 2022-03-02

### PR DESCRIPTION
**Pull request by Thomas1664**
**Describe the pull request**

- #### What does your PR fix?  
  Update to 2022-03-02
  Replace deprecated cmake functions

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Yes: Removed from CI baseline as marl has `"supports": "!uwp"` in its manifest

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

File: ports/marl/portfile.cmake [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
+    REF 9929747c9ba6354691dbacaf14f9b35433871e5b #2022-03-02
+    SHA512 454399485d292526333417474a312302aaff90cf63bc06a95c2e8b87cb92eaea547457ba3c06413e079ca29f9ea64990b9da467aeaec6ec2aa3233efddab2407
     HEAD_REF main
 )
 
-string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" MARL_BUILD_SHARED)
+set(MARL_BUILD_SHARED OFF)
+if("${VCPKG_LIBRARY_LINKAGE}" STREQUAL "dynamic")
+    set(MARL_BUILD_SHARED ON)
+endif()
```
Comment (JonLiu1993) [inline ports/marl/portfile.cmake:12]: @Thomas1664 ,Thanks for your pr, could you explain why this block needs to be modified？

Comment (Thomas1664) [reply ports/marl/portfile.cmake:12]: @JonLiu1993 This is more a style change. Post ports do it this way so I changed it.

Comment (JonLiu1993) [reply ports/marl/portfile.cmake:12]: @Thomas1664 ,The original style is what we recommend. you can refer to [this](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#choose-either-static-or-shared-binaries)

File: ports/marl/portfile.cmake [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
+    SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
         -DMARL_BUILD_SHARED=${MARL_BUILD_SHARED}
         -DMARL_INSTALL=ON
 )
 
-vcpkg_install_cmake()
+vcpkg_cmake_install()
 
+set(MARL_BUILD_SHARED OFF)
```
Comment (JonLiu1993) [inline ports/marl/portfile.cmake:23]:
```suggestion
```

File: ports/marl/portfile.cmake (hunk 1 above)
Comment (Thomas1664) [reply ports/marl/portfile.cmake:12]: Then I'll revert my change to use the old style

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**